src/dune_rules/lib.ml

```diff
@@ -1146,6 +1146,67 @@ end = struct
     ; re_exports : lib list Resolve.t
     }
 
+  module Resolved_deps_builder : sig
+    type t
+
+    val empty : t
+
+    val add_resolved : t -> resolved:lib Resolve.t -> t
+
+    val add_resolved_list : t -> resolved:lib list Resolve.t -> t
```

Comment:
`add_resolved` and `add_resolved_list` are too similar. Can we somehow make do with one of them?